docs/2_2_common_mechanisms.adoc

```diff
@@ -330,522 +343,711 @@ Set <<parameter,`parameters`>>, <<input,`inputs`>>, and <<start>> values, and re
-The red lines denote clock connections.
-The Ctrl FMU declares an output clock latexmath:[r], and the Sensor and Actuator FMUs declare input clocks, connected to latexmath:[r].
-This way, the importer knows that when the Ctrl clock latexmath:[r] ticks, then the Sensor and Actuator FMUs will be informed that the clock ticked and execute the corresponding equations.]_
+[[fmi3SerializeFMUState,`fmi3SerializeFMUState`]]
+Function <<fmi3SerializeFMUState>>::
 
-_[In general:_
+[source, C]
+----
+include::../headers/fmi3FunctionTypes.h[tags=SerializeFMUState]
+----
 
-- _The ticking of an output clock is determined by the FMU that declares it._
-_So in <<figure-synchronous-clock-scenario-clocked>>, the Supervisor FMU determines the ticking of latexmath:[s] and Ctrl FMU determines the ticking of latexmath:[r]._
-_The FMU Ctrl declared both an input clock and an outut clock, and the Sensor and Actuator FMUs both declare an input clock._
+This function serializes the data which is referenced by pointer <<FMUState>> and copies this data in to the byte vector `serializedState` of length `size`, that must be provided by the environment.
 
-- _The ticking of an input clock is signaled by the importer, which in turn will find the source that determines the ticking of such clock.]_
+[[fmi3DeSerializeFMUState,`fmi3DeSerializeFMUState`]]
+Function <<fmi3DeSerializeFMUState>>::
 
-A clocked partition is a set of equations that are associated to a <<clock>> and are executed when the corresponding <<clock>> is active.
-A clocked partition is mathematically defined as:
+[source, C]
+----
+include::../headers/fmi3FunctionTypes.h[tags=DeSerializeFMUState]
+----
 
-[latexmath]
-++++
-x_j=\begin{cases}
-    x_{j-1} & \text{if subactive}\\
-    f_j(x_{j-1},u_j,t_j) & \text{else}
-    \end{cases}
-++++
+This function deserializes the byte vector `serializedState` of length `size`, constructs a copy of the FMU state and returns <<FMUState>>, the pointer to this copy.
 
-[latexmath]
-++++
-y_j=g_j(x_{j-1},u_j,t_j);
-++++
+These functions are only supported by the FMU, if the optional capability flags <<canGetAndSetFMUState>> and <<canSerializeFMUState>> in `<fmiModelDescription><ModelExchange|CoSimulation|ScheduledExecution>` in the XML file are explicitly set to `true` (see <<fmi-for-model-exchange>>, <<fmi-for-co-simulation>>, <<fmi-for-scheduled-execution>>).
+--
 
-_[TODO: shouldn't this be:
-Claudio: rewrite this to make it clearer: the distinction between these two are just local vars and outputs.
-Claudio: Also relate to the previous example.
-Claudio: Also change the modelica sentence.
-Claudio: Make it clear that clocks are just a way to implement numerically sound event having across different FMUs.
-]_
-[latexmath]
-++++
-y_j=g_j(x_{j},u_j,t_j);
-++++
+==== Selective Computation of Variables [[selectiv-computation]]
```

Comment:
Is the selective calculation of variables as described in this subsection really a common concept or >=2 FMI interface types?
To my understanding, this is only the case in ME. 
In CS, calculations are triggerd in fmi3Dostep, fmi3NewDiscreteStates, and in ME in activateModelPartition

Reply:
I think this also applies to Step Mode in CS: it is a quality of implementation issue. If the FMU can separate state computation from output computation, why not produce outputs only when the importer requires them?
Please reopen if you do not agree.

Reply:
Do we have to state this specifically if we refer to caching? Caching by definition guarantees this. If we make an explicit comment here, this might be read as different from the standard comp-sci use. Please re-read and if you still think this is needed, please suggest exactly where to please what kind of comment.

Reply:
As a "get" can trigger calculations: I think the question is, if the FMU _may_ apply caching or _must_ apply caching.
Is it guaranteed, that you get the same result if you call a get function two times (w/o a set, DoStep, UpdateDiscrete States or activateModelPartition call inbetween).

Reply:
Changed:
This enables computation on demand.
To:
This enables computation on demand with subsequent calls returning the same value until some set operation requires re-computation.